docs/src/content/docs/request-state.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Request State
+description: Request State
+packageName: '@inox-tools/request-state'
+challenger: [Florian, Oliver]
+---
+
+Astro provides [`Astro.locals`/`context.locals`](https://docs.astro.build/en/reference/api-reference/#astrolocals) for shared state between different components being rendered in a page.
+
+This state can only be accessed from an Astro component using the `Astro` constant or from middlewares from the context. Sharing state between framework components is not provided and inconvenient.
+
+Framework components using the shared state won't have the same state on the client, resulting in problems with the server and the client rendering different contents.
+
+This library provides a solution for that, any state set using this library is shared across any component (even from UI frameworks) used in rendering the page in both server and client side.
+
+## Installing the dependency
+
+import { PackageManagers } from 'starlight-package-managers';
+
+<PackageManagers type="exec" pkg="astro" args="add @inox-tools/request-state" />
+
+## How to use
```

Comment:
I still think generics would be valuable

Reply:
Is it really any better?

```ts
getState('key') as Foo;

getState<Foo>('key');
```

You are making an assumption in both, the lib doesn't guarantee it.

Reply:
fair enough